crates/motoko/src/lib/shared.rs

```diff
@@ -64,24 +59,56 @@ impl<T> AsRef<T> for Shared<T> {
     }
 }
 
-pub trait Share<T> {
+pub trait FastClone: Clone {
+    /// A more explicit alternative to `clone()` (same use case as calling `Rc::clone(_)`).
+    fn fast_clone(&self) -> Self {
```

Comment:
But doesn't this default impl mean that every type that is `Clone` will also be `FastClone`

Doesn't that defeat the purpose of only having it be defined for those types where it is actually "fast" (O(1))?

Reply:
I just saw this pattern in another crate, and maybe it's relevant for this issue too?

https://github.com/paritytech/nohash-hasher/blob/3a8a607cb9fe20e54db2bfb3fb30fe7bbfcd6476/src/lib.rs#L185
```
pub trait IsEnabled {}

impl IsEnabled for u8 {}
impl IsEnabled for u16 {}
impl IsEnabled for u32 {}
impl IsEnabled for u64 {}
impl IsEnabled for usize {}
impl IsEnabled for i8 {}
impl IsEnabled for i16 {}
impl IsEnabled for i32 {}
impl IsEnabled for i64 {}
impl IsEnabled for isize {}

impl<T: IsEnabled> Hasher for NoHashHasher<T> {
   /* default impl, to be overridden */
}
```